libs/agno/agno/reasoning/default.py

```diff
@@ -21,6 +21,18 @@ def get_default_reasoning_agent(
 ) -> Optional["Agent"]:  # type: ignore  # noqa: F821
     from agno.agent import Agent
 
+    # Special instructions for Mistral models
+    json_instructions = ""
+    if reasoning_model.__class__.__name__ == "MistralChat":
+        json_instructions = """
+        RESPONSE FORMAT INSTRUCTIONS:
+        - Return your response as a plain JSON object.
+        - DO NOT wrap your JSON in markdown code blocks.
+        - DO NOT use ```json or ``` tags.
+        - Return ONLY the raw JSON with no other text.
+        - For example, respond with {"reasoning_steps": [...]} directly, not ```json {"reasoning_steps": [...]} ```
+        """
```

Comment:
This should not be needed, our system message does this by default if you have `use_json_mode=True`